### PR TITLE
Show Progressbar for 18MS

### LINKED ISSUE #4392: 18MS: Improved timeline
18CZ has an improved timeline. Maybe something similar could be made in 18MS?

image

### PR DESCRIPTION
**Pull request by Zwergenpunk**
Adds Progress bar for 18 MS
The train icons are in 18CZ for export, but I think it works for 18MS as well for the rusting. 
image

With OR 11 optional rule there is one more OR in the progress bar
closes #4392

### CODE REVIEW

File: lib/engine/game/g_18_ms/game.rb [hunk 1]
```diff
@@ -731,6 +731,34 @@ def president_assisted_buy(corporation, train, price)
...
+            { type: :OR, name: '6', exportAfter: true },
+            { type: :SR },
+            { type: :OR, name: '7' },
+            { type: :OR, name: '8', exportAfter: true },
+            { type: :SR },
+            { type: :OR, name: '9' },
+            { type: :OR, name: '10' },
+          ]
+
+          base_progress.push({ type: :OR, name: '11' }) if @optional_rules&.include?(:or_11)
```
Comment (tobymao) [inline lib/engine/game/g_18_ms/game.rb:758]: base_progress <<

**Comment by perwestling** [comment]
It looks pretty good but I am not sure I do agree with the use of colors.
In 18MS green phase starts when OR 2 starts. As far as I know this is the only title that use the timeline to change phase.
So I believe OR 2 should be green, and OR 1 should be yellow. 
OR3+ I am not sure what color they should have, but how about having all other boxes without color?
(You cannot know which boxes would be brown as you cannot know when the 6 train is bought, or gray when D is bought.)

The alternative is to put some icon/mark on OR2.

The overcrossed trains for rusting is fine. The alternative would be to put some icon/mark on OR4, OR6, OR8. 
And another possibility is to add 2+, 3+, 4+ but I guess that might put too much info in the timeline?

BTW, maybe END should be light blue? Like the END on the stock market in some games?

**Comment by Zwergenpunk** [comment]
I don't know how good it is to map the OR with a color to a phase, when not all phases are mappable to a OR.

I think we could add a 2+, 4+ etc to the train icon, and I also like the blue for end. Will fix that.

**Comment by Zwergenpunk** [comment]
Changed image in top comment

**Comment by scottredracecar** [comment]
I think the yellow and green should be some neutral color like white rather than one associated with a phase.

**Review by tobymao — APPROVED** [review]

**Pull request merged**